crates/api_common/src/request.rs

```diff
@@ -340,15 +340,15 @@ async fn is_image_content_type(client: &ClientWithMiddleware, url: &Url) -> Lemm
 
 /// When adding a new avatar, banner or similar image, delete the old one.
 pub async fn replace_image(
-  new_image: &Option<String>,
+  new_image: &Option<Option<DbUrl>>,
   old_image: &Option<DbUrl>,
   context: &Data<LemmyContext>,
 ) -> LemmyResult<()> {
-  if let (Some(new_image), Some(old_image)) = (new_image, old_image) {
+  if let (Some(Some(new_image)), Some(old_image)) = (new_image, old_image) {
```

Comment:
I *think* the following should work:

```suggestion
  if let Some((new_image, old_image)) = new_image.flatten().zip(old_image) {
```

Reply:
Might work, but seems less clear to me.

Reply:
Fair. Personally, I find this a bit easier than needing to parse nested `Some`s.